sources/api/netdog/src/networkd_status.rs

```diff
@@ -0,0 +1,304 @@
+//! The networkd_status module contains definitions and functions for tracking network status from networkd
+//!
+use crate::interface_id::InterfaceName;
+#[cfg(net_backend = "systemd-networkd")]
+use crate::NETWORKCTL;
+use serde::{Deserialize, Deserializer};
+#[cfg(net_backend = "systemd-networkd")]
+use snafu::{ensure, ResultExt};
+#[cfg(net_backend = "systemd-networkd")]
+use std::cmp::Ordering;
+use std::convert::TryInto;
+use std::net::IpAddr;
+#[cfg(net_backend = "systemd-networkd")]
+use std::process::Command;
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[allow(dead_code)]
+pub(crate) struct NetworkdDnsConfig {
+    #[serde(rename = "Family")]
+    family: u8,
+    #[serde(rename = "Address", deserialize_with = "ipaddr_from_vec_de")]
+    pub(crate) address: IpAddr,
+    #[serde(rename = "ConfigSource")]
+    config_source: String,
+    #[serde(rename = "ConfigProvider", deserialize_with = "ipaddr_from_vec_de")]
+    config_provider: IpAddr,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[allow(dead_code)]
+pub(crate) struct SearchDomain {
+    #[serde(rename = "Domain")]
+    pub(crate) domain: String,
+    #[serde(rename = "ConfigSource")]
+    config_source: String,
+    #[serde(rename = "ConfigProvider", deserialize_with = "ipaddr_from_vec_de")]
+    config_provider: IpAddr,
+}
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct NetworkctlIpAddr {}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+#[allow(dead_code)]
+pub(crate) struct NetworkdStatus {
+    pub(crate) name: InterfaceName,
+    #[serde(rename = "DNS")]
+    pub(crate) dns: Option<Vec<NetworkdDnsConfig>>,
+    pub(crate) search_domains: Option<Vec<SearchDomain>>,
+    #[serde(rename = "HardwareAddress")]
+    pub(crate) mac_address: Vec<u8>,
+    #[serde(rename = "Addresses", deserialize_with = "from_networkctl_addresses")]
+    pub(crate) addresses: Vec<IpAddr>,
+}
+
+// get an IpAddr from a Vec<u8> (could be 4 or 16 length)
+fn ipaddr_from_vec(address_vec: Vec<u8>) -> std::result::Result<IpAddr, Error> {
+    use error::BadIpAddressSnafu;
+    match address_vec.len() {
+        4 => {
+            let octets: [u8; 4] = address_vec.try_into().unwrap();
+            Ok(IpAddr::from(octets))
+        }
+        16 => {
+            let octets: [u8; 16] = address_vec.try_into().unwrap();
+            Ok(IpAddr::from(octets))
+        }
+        _ => BadIpAddressSnafu {
```

Comment:
Rather than an `error` import inside this function, let's just use the full path since the error module lives inside this module.